src/includes/class-alma-wc-plugin.php

```diff
@@ -752,26 +753,52 @@ function ( $key ) use ( $cart_eligibilities ) {
 	/**
 	 * Get eligibilities from cart.
 	 *
-	 * @return array<string,Eligibility>|null
+	 * @return Eligibility[]|array
 	 */
 	public function get_cart_eligibilities() {
 		if ( ! $this->eligibilities ) {
 			$alma = alma_wc_plugin()->get_alma_client();
 			if ( ! $alma ) {
-				return null;
+				return array();
 			}
 
+			$eligibility_payload_from_cart = Alma_WC_Model_Payment::get_eligibility_payload_from_cart();
 			try {
-				$this->eligibilities = $alma->payments->eligibility( Alma_WC_Model_Payment::get_eligibility_payload_from_cart() );
+				$this->eligibilities = $alma->payments->eligibility( $eligibility_payload_from_cart );
 			} catch ( RequestError $error ) {
 				$this->logger->log_stack_trace( 'Error while checking payment eligibility: ', $error );
-				return null;
+				return array();
+			}
+
+			// Fix for only having an array as results.
+			if ( is_object( $this->eligibilities ) ) {
+				$this->eligibilities = $this->format_eligibility_to_array( $eligibility_payload_from_cart, $this->eligibilities );
 			}
 		}
 
 		return $this->eligibilities;
 	}
 
+	/**
+	 * Fix for only having an array as results (see Alma\API\Endpoints\Payments:eligibility())
+	 *
+	 * @param array       $eligibility_payload_from_cart Payment data to check the eligibility for.
+	 * @param Eligibility $eligibility The eligibility.
+	 *
+	 * @return array
+	 */
+	protected function format_eligibility_to_array( array $eligibility_payload_from_cart, $eligibility ) {
+		$is_v1_payload = array_key_exists( 'payment', $eligibility_payload_from_cart );
+		$result = array();
+
+		if ( $is_v1_payload ) {
+			$result[ $eligibility->getInstallmentsCount() ] = $eligibility;
+		} else {
+			$result[ $eligibility->getPlanKey() ] = $eligibility;
+		}
+
+		return $result;
```

Comment:
```suggestion
		return [ $eligibility->getPlanKey() => $eligibility ];
```
in fact, we don't call eligibility v1 from our plugin ... so we are sur to get eligibility v2 formatted array... the only thing you have to do here is to set `getPlanKey` as array key

Maybe you could delegate the `is_object` test to the current method instead of doing it into `get_cart_eligibilities`

Reply:
I get rid of the check v1, i still prefer not to move le is_object elsewhere it will be more if/else return

Reply:
however, I think this test on `is_object` is fully part of the format method because the format depends on the `is_object` (so `is_object` seems to be a responsibility of the format method)